ros2-library/src/main/java/us/ihmc/ros2/ROS2NodeBuilder.java

```diff
@@ -0,0 +1,438 @@
+       */
+      UDPV4_ONLY,
+      /**
+       * INTRAPROCESS_ONLY attempts to force Publishers to directly call reception functions of Subscribers.
+       * This mode still enables a shared memory (SHM) transport, as there has to be one fallback transport on a Participant.
+       * This mode is especially useful for unit testing and single-process applications.
+       * This mode does not bind to any network address, unless SHM transport was unable to initialize.
+       * This mode is NOT compatible with mixed-transports, e.g. a participant with SHM and UDPv4 transports.
+       * Documentation: <a href="https://fast-dds.docs.eprosima.com/en/v2.14.3/fastdds/transport/intraprocess.html#intraprocess-delivery">Intra-process
+       * delivery</a>
+       */
+      INTRAPROCESS_ONLY
+   }
+
+   private int domainId = UNSET_DOMAIN_ID;
+   private String namespace = DEFAULT_NAMESPACE;
+   private boolean useSharedMemory = true;
+   private InetAddress[] addressRestriction = null;
+   private boolean parseEnvironment = true;
+   private boolean parseProperties = true;
+   private boolean parseNetworkParametersConfig = true;
+   @Nullable
+   private SpecialTransportMode specialTransportMode;
+
+   private final transient StringJoiner buildPrintout = new StringJoiner("\n\t\t");
+
+   public ROS2NodeBuilder domainId(int domainId)
+   {
+      this.domainId = domainId;
+      return this;
+   }
+
+   public ROS2NodeBuilder namespace(String namespace)
+   {
+      this.namespace = namespace;
+      return this;
+   }
+
+   public ROS2NodeBuilder useSharedMemory(boolean useSharedMemory)
+   {
+      this.useSharedMemory = useSharedMemory;
+      return this;
+   }
+
+   public ROS2NodeBuilder addressRestriction(InetAddress... addressRestriction)
+   {
+      this.addressRestriction = addressRestriction;
+      return this;
+   }
+
+   public ROS2NodeBuilder parseEnvironment(boolean parseEnvironment)
+   {
+      this.parseEnvironment = parseEnvironment;
+      return this;
+   }
+
+   public ROS2NodeBuilder parseProperties(boolean parseProperties)
+   {
+      this.parseProperties = parseProperties;
+      return this;
+   }
+
+   public ROS2NodeBuilder parseNetworkParametersConfig(boolean parseNetworkParametersConfig)
+   {
+      this.parseNetworkParametersConfig = parseNetworkParametersConfig;
+      return this;
+   }
+
+   public ROS2NodeBuilder specialTransportMode(@Nullable SpecialTransportMode specialTransportMode)
+   {
+      this.specialTransportMode = specialTransportMode;
+      return this;
+   }
+
+   public ROS2Node build(String name)
+   {
+      buildPrintout.add("Building ROS2Node: " + name);
+      return new ROS2Node(name, namespace, buildProfile(), specialTransportMode);
+   }
+
+   public RealtimeROS2Node buildRealtime(String name)
+   {
+      return buildRealtime(name, new PeriodicNonRealtimeThreadSchedulerFactory());
```

Comment:
Any idea why the default factory for a RealtimeROS2Node is NON realtime? I know this is how ROS2Tools did it too, but it seems so odd.

Reply:
Tbh I don't know. I am just gonna leave it how it was